File: check_dependencies/check_dependencies.py
import subprocess

dependencies = {
    "Ruby": "ruby -v",
    "Bundler": "bundle -v",
}

for name, command in dependencies.items():
    try:
        output = subprocess.check_output(command, shell=True, stderr=subprocess.DEVNULL).decode().strip()
        print(f"{name}: {output}")
    except subprocess.CalledProcessError:
        print(f"{name}: Not installed")

File: check_dependencies/check_dependencies.rb
require 'open3'

dependencies = {
  "Ruby" => "ruby -v",
  "Bundler" => "bundle -v",
}

dependencies.each do |name, command|
  output, status = Open3.capture2(command)
  if status.success?
    puts "#{name}: #{output.strip}"
  else
    puts "#{name}: Not installed"
  end
end